### PR TITLE
chore: improve server stream typings

### PR DESCRIPTION
**Pull request by ggazzo**
https://rocketchat.atlassian.net/browse/ARCH-1110

### CODE REVIEW

File: apps/meteor/client/startup/incomingMessages.ts [hunk 1]
```diff
@@ -20,16 +20,16 @@ Meteor.startup(() => {
...
 				{
 					rid: sub.rid,
-					...(sub?.ignored ? { 'u._id': { $nin: sub.ignored } } : { ignored: { $exists: true } }),
+					...('ignored' in sub ? { 'u._id': { $nin: sub.ignored } } : { ignored: { $exists: true } }),
 				},
 				{ $unset: { ignored: true } },
 				{ multi: true },
 			);
-			if (sub?.ignored) {
+			if ('ignored' in sub) {
```
Comment (ggazzo) [inline apps/meteor/client/startup/incomingMessages.ts:32]:
```suggestion
			if ('ignored' in sub && sub.ignored) {
```

File: apps/meteor/client/startup/incomingMessages.ts [hunk 2]
```diff
@@ -20,16 +20,16 @@ Meteor.startup(() => {
 	});
 
 	CachedCollectionManager.onLogin(() => {
-		Notifications.onUser('subscriptions-changed', (_action: unknown, sub: ISubscription) => {
+		Notifications.onUser('subscriptions-changed', (_action, sub) => {
 			ChatMessage.update(
 				{
 					rid: sub.rid,
-					...(sub?.ignored ? { 'u._id': { $nin: sub.ignored } } : { ignored: { $exists: true } }),
+					...('ignored' in sub ? { 'u._id': { $nin: sub.ignored } } : { ignored: { $exists: true } }),
```
Comment (ggazzo) [inline apps/meteor/client/startup/incomingMessages.ts:27]:
```suggestion
					...('ignored' in sub && sub.ignored ? { 'u._id': { $nin: sub.ignored } } : { ignored: { $exists: true } }),
```

File: apps/meteor/app/authorization/server/methods/removeUserFromRole.ts [hunk 3]
```diff
@@ -11,12 +11,12 @@ import { apiDeprecationLogger } from '../../../lib/server/lib/deprecationWarning
...
 	// eslint-disable-next-line @typescript-eslint/naming-convention
 	interface ServerMethods {
-		'authorization:removeUserFromRole'(roleId: IRole['_id'], username: IUser['username'], scope: undefined): Promise<boolean>;
+		'authorization:removeUserFromRole'(roleId: IRole['_id'], username: IUser['username'], scope?: string): Promise<boolean>;
 	}
 }
 
 Meteor.methods<ServerMethods>({
-	async 'authorization:removeUserFromRole'(roleId, username, scope) {
+	async 'authorization:removeUserFromRole'(roleId, username, scope = 'Users') {
```
Comment (ggazzo) [inline apps/meteor/app/authorization/server/methods/removeUserFromRole.ts:19]:
```suggestion
	async 'authorization:removeUserFromRole'(roleId, username, scope) {
```

File: apps/meteor/app/livechat/client/lib/stream/queueManager.ts [hunk 4]
```diff
@@ -54,14 +51,20 @@ const updateInquiries = async (inquiries: ILivechatInquiryRecord[] = []) =>
 	inquiries.forEach((inquiry) => LivechatInquiry.upsert({ _id: inquiry._id }, { ...inquiry, _updatedAt: new Date(inquiry._updatedAt) }));
 
 const getAgentsDepartments = async (userId: IOmnichannelAgent['_id']) => {
-	const { departments } = await sdk.rest.get(`/v1/livechat/agents/${userId}/departments`, { enabledDepartmentsOnly: 'true' });
+	const { departments } = await APIClient.get(`/v1/livechat/agents/${userId}/departments`, { enabledDepartmentsOnly: 'true' });
```
Comment (sampaiodiego) [inline apps/meteor/app/livechat/client/lib/stream/queueManager.ts:54]: any reason to not use `sdk.rest` here?

File: apps/meteor/server/modules/notifications/notifications.module.ts [hunk 5]
```diff
@@ -477,47 +482,98 @@ export class NotificationsModule {
...
+	P extends string,
+	E extends StreamKeys<T> = StreamKeys<T>,
+> = E extends `${infer X}/${infer I}` ? (P extends X ? I : never) : never;
+
+// type X = ExtractNotifyUserEventName<'notify-user', 'webrtc'>;
+
+// type Y = StreamerCallbackArgs<'notify-user', `${string}/${'webrtc'}`>;
+// const a: NotificationsModule = {};
+
+// a.notifyUserInThisInstance('asda', 'webrtc', 'a', 'b', 'c');
```
Comment (sampaiodiego) [inline apps/meteor/server/modules/notifications/notifications.module.ts:579]:
```suggestion
```

File: apps/meteor/server/modules/watchers/watchers.module.ts [hunk 6]
```diff
@@ -147,14 +198,25 @@ export function initWatchers(watcher: DatabaseWatcher, broadcast: BroadcastCallb
...
+		if (clientAction === 'removed') {
+			void broadcast('watch.roles', {
+				clientAction: 'removed',
+				role: {
+					_id: id,
+					name: id,
+				},
+			});
+			return;
+		}
```
Comment (sampaiodiego) [inline apps/meteor/server/modules/watchers/watchers.module.ts:212]:
```suggestion
		if (clientAction === 'removed') {
			void broadcast('watch.roles', {
				clientAction: 'removed',
				role: {
					_id: id,
					name: id,
				},
			});
			return;
		}

		const role = data || (await Roles.findOneById(id));
```

File: apps/meteor/app/livechat/client/lib/stream/queueManager.ts (hunk 4 above)
Comment (sampaiodiego) [inline apps/meteor/app/livechat/client/lib/stream/queueManager.ts:54]:
```suggestion
	const { departments } = await sdk.rest.get(`/v1/livechat/agents/${userId}/departments`, { enabledDepartmentsOnly: 'true' });
```

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
look files

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**